### DISCUSSION TITLE
Modern interactive XAML

### DISCUSSION THREAD

**Discussion by GroophyLifefor** [dotnet/wpf#7613] [Ideas]
In my experience, WPF is awesome, and I'll default on my freelance career, but some interactive thing can make easier to make ui.
like:
```xml
window.Width="900"
---
<Grid Width="300"  HorizontalAlignment="Left" Background="Blue"></Grid>
<Grid Width="300"  HorizontalAlignment="Center" Background="DodgerBlue"></Grid>
<Grid Width="300"  HorizontalAlignment="Right" Background="DarkBlue"></Grid>
```
![image](https://user-images.githubusercontent.com/77299279/224932115-e455596d-bed1-4c9a-827c-bc43b37411fe.png)
`but It will broken in resize, yes I can fix with same code but I think It can easier.`

```xml
window.Width="900"
---
<Grid Width="{window.Width/3}"  HorizontalAlignment="Left" Background="Blue"></Grid>
<Grid Width="{window.Width/3}"  HorizontalAlignment="Center" Background="DodgerBlue"></Grid>
<Grid Width="{window.Width/3}"  HorizontalAlignment="Right" Background="DarkBlue"></Grid>
```

Actualy I want to say inline executment.

**Comment by haefele** [comment]
You can already do that with the WPF `Grid`.

```xaml
<Grid>
    <Grid.ColumnDefinitions>
        <ColumnDefinition Width="*" />
        <ColumnDefinition Width="*" />
        <ColumnDefinition Width="*" />
    </Grid.ColumnDefinitions>

    <Grid Grid.Column="0" Background="Blue"></Grid>
    <Grid Grid.Column="1" Background="DodgerBlue"></Grid>
    <Grid Grid.Column="2" Background="DarkBlue"></Grid>
</Grid>
```

**Reply by GroophyLifefor** [reply]
@haefele The event is not design, please read my article more carefully, I wrote an article aimed at code that can be written in xaml that can hot-reload.
```
Width="{window. Width/3}"
```

**Reply by miloush** [reply]
You can probably write a markup extension to interpret a string parameter as C# (but you could also just bind it to a view model).

**Reply by GroophyLifefor** [reply]
@miloush Thanks for suggestions ❤️
